File: AmneziaCore/Models/DnsProvider.swift
import Foundation

public enum DnsProvider: String, CaseIterable, Identifiable, Codable, Sendable {
    case serverDefault = "Server Default"
    case cloudflare = "Cloudflare (1.1.1.1)"
    case adguard = "AdGuard (Blocks Ads & Trackers)"
    case quad9 = "Quad9 (Malware Protection)"
    case custom = "Custom DNS"

    public var id: String { rawValue }

    public var ips: [String] {
        switch self {
        case .serverDefault: return []
        case .cloudflare: return ["1.1.1.1", "1.0.0.1"]
        case .adguard: return ["94.140.14.14", "94.140.15.15"]
        case .quad9: return ["9.9.9.9", "149.112.112.112"]
        case .custom: return []
        }
    }
}

File: AmneziaAndroid/core/src/main/java/org/amnezia/core/models/DnsProvider.kt
package org.amnezia.core.models

enum class DnsProvider(val title: String, val subtitle: String, val servers: List<String>) {
    SERVER_DEFAULT("Server Default", "Use the DNS servers from the imported config", emptyList()),
    CLOUDFLARE("Cloudflare", "Fast & privacy-oriented (1.1.1.1)", listOf("1.1.1.1", "1.0.0.1")),
    GOOGLE("Google Public DNS", "High reliability & global reach (8.8.8.8)", listOf("8.8.8.8", "8.8.4.4")),
    QUAD9("Quad9", "Malware blocking & secure (9.9.9.9)", listOf("9.9.9.9", "149.112.112.112")),
    ADGUARD("AdGuard DNS", "Built-in ad & tracker blocking", listOf("94.140.14.14", "94.140.15.15")),
    CUSTOM("Custom DNS", "User defined addresses", emptyList())
}
